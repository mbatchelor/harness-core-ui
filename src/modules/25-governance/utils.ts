import type { GovernanceMetadata } from 'services/cd-ng'

interface DoesGovernanceMetaDataHasErrorOrWarningRtn {
  governanceMetaDataHasError: boolean
  governanceMetaDataHasWarning: boolean
}
export const doesGovernanceHasErrorOrWarning = (
  governanceMetadata: GovernanceMetadata | undefined
): DoesGovernanceMetaDataHasErrorOrWarningRtn => {
  let governanceMetaDataHasWarning = false
  let governanceMetaDataHasError = false
  if (governanceMetadata && governanceMetadata?.status === 'error') {
    governanceMetaDataHasError = true
  }
  if (governanceMetadata && governanceMetadata?.status === 'warning') {
    governanceMetaDataHasWarning = true
  }
  return { governanceMetaDataHasError, governanceMetaDataHasWarning }
}
