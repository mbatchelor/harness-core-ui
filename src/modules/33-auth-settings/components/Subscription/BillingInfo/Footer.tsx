/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { Layout, Text, Button, ButtonVariation } from '@harness/uicore'
import { useStrings } from 'framework/strings'
import { SubscribeViews } from '@common/constants/SubscriptionTypes'
import { TIME_TYPE } from '@auth-settings/pages/subscriptions/plans/planUtils'
import css from '@auth-settings/modals/Subscription/useSubscriptionModal.module.scss'

interface FooterProps {
  time: TIME_TYPE
  setView: (view: SubscribeViews) => void
}

export const Footer = ({ time, setView }: FooterProps): React.ReactElement => {
  const { getString } = useStrings()
  const timeDescr = time === TIME_TYPE.MONTHLY ? getString('common.perMonth') : getString('common.perYear')

  function handleNext(): void {
    setView(SubscribeViews.FINALREVIEW)
  }

  function handleBack(): void {
    setView(SubscribeViews.CALCULATE)
  }

  return (
    <Layout.Horizontal className={css.footer}>
      <Layout.Horizontal spacing={'small'}>
        <Button variation={ButtonVariation.SECONDARY} onClick={handleBack} icon="chevron-left">
          {getString('back')}
        </Button>
        <Button variation={ButtonVariation.PRIMARY} onClick={handleNext} rightIcon="chevron-right">
          {getString('authSettings.billing.next')}
        </Button>
      </Layout.Horizontal>
      <Layout.Vertical>
        <Text>{`${getString('authSettings.costCalculator.payingToday')} ${timeDescr}`}</Text>
        <Text>{getString('authSettings.plusTax')}</Text>
      </Layout.Vertical>
    </Layout.Horizontal>
  )
}
