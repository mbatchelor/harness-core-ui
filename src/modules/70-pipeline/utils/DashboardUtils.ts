/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { TooltipFormatterContextObject } from 'highcharts'

interface PointStats {
  deployments?: {
    failure?: number
    success?: number
    total?: number
  }
  time?: number
}

/* istanbul ignore next */
export const renderTooltipContent = ({
  time,
  failureRate,
  count,
  successCount,
  failureCount
}: {
  time: string | number
  failureRate: string | number
  count?: number
  successCount?: number
  failureCount?: number
}) => {
  return `<div style="padding: 16px; color: white; width: 282px; height: 128px;">
        <div style="display: flex; justify-content: space-between; border-bottom: 0.5px solid rgba(243, 243, 250); padding-bottom: 7px; margin-bottom: 15px;">
          <div style="font-weight: normal; font-size: 12px; line-height: 18px; opacity: 0.8;">${time}</div>
          <div>
            <span style="white-space: pre; font-weight: bold; font-size: 12px; line-height: 18px; opacity: 0.8;">Deployments: </span>
            <span style="font-weight: bold; font-size: 12px; line-height: 18px;">${count}</span>
          </div>
        </div>
        <div style="display: flex; justify-content: space-between;">
          <div>
            <p style="font-weight: 500; font-size: 10px; line-height: 14px; letter-spacing: 0.2px; color: #D9DAE6; margin-bottom: 0px;">Failure Rate</p>
            <p style="font-weight: 600; font-size: 28px; line-height: 38px; color: #FBE6E4;">${failureRate}</p>
          </div>
          <div style="margin-right: 8px;">
            <div style="display: flex; align-items: center; margin-bottom: 6px;">
              <div style="height: 6px; width: 12px; background-color: #5FB34E; border-radius: 16px; display: inline-block; margin-right: 8px"></div>
              <span style="white-space: pre; font-weight: bold; font-size: 12px; line-height: 16px; letter-spacing: 0.2px; opacity: 0.8;">Success </span>
              <span style="font-weight: bold; font-size: 12px; line-height: 16px; letter-spacing: 0.2px;">${successCount}</span>
            </div>
            <div style="display: flex; align-items: center;">
              <div style="height: 6px; width: 12px; background-color: #EE5F54; border-radius: 16px; display: inline-block; margin-right: 8px"></div>
              <span style="white-space: pre; font-weight: bold; font-size: 12px; line-height: 18px; opacity: 0.8;">Failed </span>
              <span style="font-weight: bold; font-size: 12px; line-height: 18px;">${failureCount}</span>
            </div>
          </div>
        </div>
      </div>`
}

/* istanbul ignore next */
export const getTooltip = (currPoint: TooltipFormatterContextObject): string => {
  const custom = currPoint?.series?.userOptions?.custom
  const point: PointStats = custom?.[currPoint.key]
  const time =
    point && point?.time
      ? new Date(point?.time).toLocaleDateString('en-US', { day: 'numeric', month: 'long' })
      : currPoint.x
  let failureRate: string | number = 'Infinity'
  if (point?.deployments?.failure && point.deployments?.total) {
    failureRate = ((point.deployments.failure / point.deployments.total) * 100).toFixed(1) + '%'
  }
  if (point?.deployments?.failure === 0) {
    failureRate = '0'
  }
  return renderTooltipContent({
    time,
    failureRate,
    count: point?.deployments?.total,
    successCount: point?.deployments?.success,
    failureCount: point?.deployments?.failure
  })
}
