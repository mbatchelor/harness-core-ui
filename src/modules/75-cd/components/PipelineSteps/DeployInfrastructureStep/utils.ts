/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { FormikProps } from 'formik'
import { isEmpty } from 'lodash-es'

import { getMultiTypeFromValue, MultiTypeInputType } from '@harness/uicore'
import type { EnvironmentResponseDTO, PipelineInfrastructure } from 'services/cd-ng'

import type { StepViewType } from '@pipeline/components/AbstractSteps/Step'

export interface DeployInfrastructureProps {
  initialValues: PipelineInfrastructure
  onUpdate?: (data: PipelineInfrastructure) => void
  readonly: boolean
  allowableTypes: MultiTypeInputType[]
  stepViewType?: StepViewType
  inputSetData?: {
    template?: PipelineInfrastructure
    path?: string
    readonly?: boolean
  }
}

export interface DeployInfrastructureState {
  isEdit: boolean
  isEnvironment: boolean
  formik?: FormikProps<PipelineInfrastructure>
  data?: EnvironmentResponseDTO
}

export function isEditEnvironment(data: PipelineInfrastructure): boolean {
  if (getMultiTypeFromValue(data.environmentRef) !== MultiTypeInputType.RUNTIME && !isEmpty(data.environmentRef)) {
    return true
  } else if (data.environment && !isEmpty(data.environment.identifier)) {
    return true
  }
  return false
}

export function isEditInfrastructure(data: any): boolean {
  if (
    getMultiTypeFromValue(data.infrastructureRef) !== MultiTypeInputType.RUNTIME &&
    !isEmpty(data.infrastructureRef)
  ) {
    return true
  } else if (data.infrastructure && !isEmpty(data.infrastructure.identifier)) {
    return true
  }
  return false
}
