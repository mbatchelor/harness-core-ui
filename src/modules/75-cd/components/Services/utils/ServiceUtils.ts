/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { set } from 'lodash-es'
import type { PipelineInfoConfig, ServiceDefinition } from 'services/cd-ng'

export enum ServiceTabs {
  SUMMARY = 'summaryTab',
  Configuration = 'configuration',
  REFERENCED_BY = 'refrencedByTab',
  ActivityLog = 'activityLog'
}

export const DefaultNewStageName = 'Stage Name'
export const DefaultNewStageId = 'stage_id'
export const DefaultNewServiceId = '-1'

export const newServiceState = {
  service: {
    name: '',
    identifier: '',
    description: '',
    gitOpsEnabled: false,
    tags: {},
    serviceDefinition: {
      type: '' as ServiceDefinition['type'],
      spec: {}
    }
  }
}

const DefaultService = {
  serviceDefinition: {
    spec: {}
  },
  gitOpsEnabled: false
}

export const initialServiceState = {
  service: { ...DefaultService }
  // isLoading: false,
  // isBETemplateUpdated: false,
  // isDBInitialized: false,
  // isUpdated: false,
  // isInitialized: false,
  // gitDetails: {},
  // entityValidityDetails: {}
}

export const setNameIDDescription = (draftData: PipelineInfoConfig, updatedData: PipelineInfoConfig): void => {
  set(draftData, 'identifier', updatedData.identifier)
  set(draftData, 'name', updatedData.name)
  set(draftData, 'description', updatedData.description)
  set(draftData, 'tags', updatedData.tags)
}
