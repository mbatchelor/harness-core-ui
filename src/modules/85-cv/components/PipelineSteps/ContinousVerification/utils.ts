/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { getMultiTypeFromValue, MultiTypeInputType, RUNTIME_INPUT_VALUE, SelectOption } from '@wings-software/uicore'
import { isEmpty, isNull, isUndefined, omit, omitBy, set } from 'lodash-es'
import { FormikErrors, yupToFormErrors } from 'formik'
import * as Yup from 'yup'
import type { UseStringsReturn } from 'framework/strings'
import { getDurationValidationSchema } from '@common/components/MultiTypeDuration/MultiTypeDuration'
import type { StringsMap } from 'stringTypes'
import type { ContinousVerificationData, spec, VerifyStepMonitoredService } from './types'
import {
  VerificationSensitivityOptions,
  durationOptions,
  baseLineOptions,
  trafficSplitPercentageOptions,
  SensitivityTypes,
  defaultMonitoredServiceSpec,
  monitoredServiceRefPath
} from './constants'
import { MONITORED_SERVICE_TYPE } from './components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/SelectMonitoredServiceType/SelectMonitoredServiceType.constants'

/**
 * checks if a field is a runtime input.
 * @param field
 * @returns boolean
 */
export function checkIfRunTimeInput(field: string | SelectOption | number | undefined): boolean {
  return getMultiTypeFromValue(field as string) === MultiTypeInputType.RUNTIME
}

/**
 * validates a particular field for errors.
 * @param fieldValue
 * @param fieldKey
 * @param data
 * @param errors
 * @param getString
 * @param isRequired
 */
export function validateField(
  fieldValue: string,
  fieldKey: string,
  data: ContinousVerificationData,
  errors: any,
  getString: UseStringsReturn['getString'],
  isRequired = true
): void {
  if (checkIfRunTimeInput(fieldValue) && isRequired && isEmpty(data?.spec?.spec && data?.spec?.spec[fieldKey])) {
    set(errors, `spec.spec.${fieldKey}`, getString('fieldRequired', { field: fieldKey }))
  }
}

export function validateMonitoredService(
  data: ContinousVerificationData,
  errors: FormikErrors<ContinousVerificationData>,
  getString: (key: keyof StringsMap, vars?: Record<string, any> | undefined) => string,
  isRequired: boolean,
  monitoredService?: VerifyStepMonitoredService
): void {
  if (
    checkIfRunTimeInput(monitoredService?.spec?.monitoredServiceRef) &&
    isRequired &&
    isEmpty(data?.spec?.monitoredService?.spec?.monitoredServiceRef)
  ) {
    set(errors, monitoredServiceRefPath, getString('fieldRequired', { field: 'Monitored service' }))
  }
}

/**
 * validates timeout field.
 * @param template
 * @param getString
 * @param data
 * @param errors
 */
export function validateTimeout(
  getString: UseStringsReturn['getString'],
  data: ContinousVerificationData,
  errors: any,
  template?: ContinousVerificationData,
  isRequired = true
): void {
  if (checkIfRunTimeInput(template?.timeout)) {
    let timeout: Yup.ObjectSchema
    if (isRequired) {
      timeout = Yup.object().shape({
        timeout: getDurationValidationSchema({ minimum: '10s' }).required(getString?.('validation.timeout10SecMinimum'))
      })
    } else {
      timeout = Yup.object().shape({
        timeout: getDurationValidationSchema({ minimum: '10s' })
      })
    }

    try {
      timeout.validateSync(data)
    } catch (e) {
      if (e instanceof Yup.ValidationError) {
        const err = yupToFormErrors(e)
        Object.assign(errors, err)
      }
    }
  }
}

/**
 * returns yaml data for spec field.
 * @param specInfo
 * @returns spec
 */
export function getSpecYamlData(specInfo?: spec, type?: string): spec {
  let validspec = omitBy(specInfo, v => isUndefined(v) || isNull(v) || v === '')
  switch (type) {
    case 'LoadTest':
      validspec = omit(validspec, ['trafficsplit'])
      break
    case 'Bluegreen':
    case 'Canary':
      validspec = omit(validspec, ['baseline'])
      break
    default:
      break
  }

  Object.keys(validspec).map((key: string) => {
    //TODO logic in if block will be removed once backend api is fixed : https://harness.atlassian.net/browse/CVNG-2481
    if (key === 'sensitivity') {
      validspec[key] = validspec[key].value
        ? SensitivityTypes[validspec[key].value as keyof typeof SensitivityTypes]
        : validspec[key]
    } else {
      validspec[key] = validspec[key].value ? validspec[key].value : validspec[key]
    }
  })

  return validspec
}

export function getMonitoredServiceYamlData(spec: ContinousVerificationData['spec']): VerifyStepMonitoredService {
  let monitoredService: VerifyStepMonitoredService = defaultMonitoredServiceSpec

  switch (spec?.monitoredService?.type) {
    case MONITORED_SERVICE_TYPE.DEFAULT:
      monitoredService = defaultMonitoredServiceSpec
      break
    case MONITORED_SERVICE_TYPE.CONFIGURED:
      monitoredService = {
        type: MONITORED_SERVICE_TYPE.CONFIGURED,
        spec: {
          monitoredServiceRef: getMonitoredServiceRef(spec)
        }
      }
      break
    default:
      monitoredService = defaultMonitoredServiceSpec
  }
  return monitoredService
}

export function getMonitoredServiceRef(spec: ContinousVerificationData['spec']): string {
  return spec?.monitoredService?.spec?.monitoredServiceRef as string
}

/**
 * returns forms data for spec field.
 * @param specInfo
 * @returns spec
 */
export function getSpecFormData(specInfo: spec | undefined): spec {
  const validspec: spec | undefined = { ...specInfo }
  if (specInfo) {
    Object.keys(specInfo).map((key: string) => {
      switch (key) {
        case 'sensitivity':
          setFieldData(validspec, 'sensitivity', VerificationSensitivityOptions)
          break
        case 'duration':
          setFieldData(validspec, 'duration', durationOptions)
          break
        case 'baseline':
          setFieldData(validspec, 'baseline', baseLineOptions)
          break
        case 'trafficsplit':
          setFieldData(validspec, 'trafficsplit', trafficSplitPercentageOptions)
          break
        default:
      }
    })
  }
  return validspec
}

/**
 * sets particular field data
 * @param validspec
 * @param field
 * @param options
 */
export function setFieldData(validspec: spec | undefined, field: string, fieldOptions: SelectOption[]): void {
  //finding the complete option if the field is fixed input
  if (validspec && validspec[field] && validspec[field] !== RUNTIME_INPUT_VALUE) {
    //TODO logic in if block will be removed once backend api is fixed : https://harness.atlassian.net/browse/CVNG-2481
    if (field === 'sensitivity') {
      validspec[field] = fieldOptions.find(
        (el: SelectOption) =>
          SensitivityTypes[el.value as keyof typeof SensitivityTypes] === (validspec && validspec[field])
      )
    } else {
      validspec[field] = fieldOptions.find((el: SelectOption) => el.value === (validspec && validspec[field]))
    }
  }
}
