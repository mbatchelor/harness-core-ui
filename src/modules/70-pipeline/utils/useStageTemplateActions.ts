/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { defaultTo } from 'lodash-es'
import { parse } from 'yaml'
import produce from 'immer'
import { useCallback } from 'react'
import type { StageElementConfig } from 'services/cd-ng'
import { usePipelineContext } from '@pipeline/components/PipelineStudio/PipelineContext/PipelineContext'
import { createTemplate, getStageType } from '@pipeline/utils/templateUtils'
import type { TemplateSummaryResponse } from 'services/template-ng'

interface TemplateActionsReturnType {
  addOrUpdateTemplate: (selectedTemplate?: TemplateSummaryResponse) => Promise<void>
  removeTemplate: () => Promise<void>
}

export function useStageTemplateActions(): TemplateActionsReturnType {
  const {
    state: {
      selectionState: { selectedStageId = '' },
      templateTypes
    },
    updateStage,
    getStageFromPipeline,
    getTemplate
  } = usePipelineContext()
  const { stage } = getStageFromPipeline(selectedStageId)

  const addOrUpdateTemplate = useCallback(
    async (selectedTemplate?: TemplateSummaryResponse) => {
      try {
        const { template, isCopied } = await getTemplate({
          templateType: 'Stage',
          selectedChildType: getStageType(stage?.stage, templateTypes),
          selectedTemplate
        })
        const node = stage?.stage
        const processNode = isCopied
          ? produce(defaultTo(parse(template?.yaml || '')?.template.spec, {}) as StageElementConfig, draft => {
              draft.name = defaultTo(node?.name, '')
              draft.identifier = defaultTo(node?.identifier, '')
            })
          : createTemplate(node, template)
        await updateStage(processNode)
      } catch (_) {
        // Do nothing.. user cancelled template selection
      }
    },
    [stage?.stage, templateTypes, getTemplate, updateStage]
  )

  const removeTemplate = useCallback(async () => {
    const node = stage?.stage
    const processNode = produce({} as StageElementConfig, draft => {
      draft.name = defaultTo(node?.name, '')
      draft.identifier = defaultTo(node?.identifier, '')
      draft.type = getStageType(node, templateTypes)
    })
    await updateStage(processNode)
  }, [stage?.stage, templateTypes, updateStage])

  return { addOrUpdateTemplate, removeTemplate }
}
