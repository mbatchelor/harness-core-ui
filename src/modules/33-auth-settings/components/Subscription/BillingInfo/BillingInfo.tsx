/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { Layout } from '@harness/uicore'
import type { Module } from 'framework/types/ModuleName'
import type { SubscribeViews } from '@common/constants/SubscriptionTypes'
import { useStrings } from 'framework/strings'
import type { TIME_TYPE } from '@auth-settings/pages/subscriptions/plans/planUtils'
import { Footer } from './Footer'
import { Header } from '../Header'

interface BillingInfoProps {
  module: Module
  setView: (view: SubscribeViews) => void
  time: TIME_TYPE
}
export const BillingInfo = ({ module, setView, time }: BillingInfoProps): React.ReactElement => {
  const { getString } = useStrings()
  return (
    <Layout.Vertical>
      <Header module={module} stepDescription={getString('authSettings.billing.step')} step={2} />
      <Footer setView={setView} time={time} />
    </Layout.Vertical>
  )
}
