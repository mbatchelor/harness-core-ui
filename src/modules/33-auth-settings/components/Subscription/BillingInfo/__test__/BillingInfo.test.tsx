/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render, fireEvent, act, waitFor } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import { SubscribeViews } from '@common/constants/SubscriptionTypes'
import { TIME_TYPE } from '@auth-settings/pages/subscriptions/plans/planUtils'
import { BillingInfo } from '../BillingInfo'

describe('BillingInfo', () => {
  test('render', () => {
    const { container, getByText } = render(
      <TestWrapper>
        <BillingInfo module="cf" setView={jest.fn()} time={TIME_TYPE.MONTHLY} />
      </TestWrapper>
    )
    expect(getByText('authSettings.billing.step')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  test('footer', async () => {
    const setViewMock = jest.fn()
    const { getByText } = render(
      <TestWrapper>
        <BillingInfo module="cf" setView={setViewMock} time={TIME_TYPE.YEARLY} />
      </TestWrapper>
    )
    act(() => {
      fireEvent.click(getByText('back'))
    })
    await waitFor(() => {
      expect(setViewMock).toBeCalledWith(SubscribeViews.CALCULATE)
    })
    act(() => {
      fireEvent.click(getByText('authSettings.billing.next'))
    })
    await waitFor(() => {
      expect(setViewMock).toBeCalledWith(SubscribeViews.FINALREVIEW)
    })
  })
})
