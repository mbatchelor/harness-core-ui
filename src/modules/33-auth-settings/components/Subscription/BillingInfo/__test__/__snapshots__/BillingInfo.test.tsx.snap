// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BillingInfo render 1`] = `
<div>
  <div
    class="Layout--vertical StyledProps--main"
  >
    <div
      class="Layout--vertical StyledProps--main StyledProps--padding-bottom-large"
    >
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-variation-h4 StyledProps--padding-bottom-large"
        style="display: flex; align-items: center;"
      >
        <span
          class="bp3-icon StyledProps--main Text--icon StyledProps--padding-right-small"
          data-icon="ff-solid"
        >
          <svg
            fill="none"
            height="24"
            viewBox="0 0 32 32"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <lineargradient
              gradientUnits="userSpaceOnUse"
              id="ff-solid_svg__a"
              x1="6.837"
              x2="23.78"
              y1="2"
              y2="28.227"
            >
              <stop
                offset="0"
                stop-color="#ffc100"
              />
              <stop
                offset="1"
                stop-color="#e66b35"
              />
            </lineargradient>
            <g
              fill="url(#ff-solid_svg__a)"
            >
              <path
                d="M10.632 2.08A5.19 5.19 0 004.62 6.29L2.08 20.697a5.19 5.19 0 004.21 6.012l7.463 1.316a.2.2 0 00.208-.298L6.882 15.656c-.296-.469-.292-1.382.05-1.6.08-.05.313-.259.313-.259l.398-.436.066-.072c.37-.406.704-.773 1.76-1.302l.227-.079c1.548-.484 2.41-.449 3.27-.414.862.035 1.724.07 3.274-.414 1.272-.398 2.43-1.178 3.074-1.667a.622.622 0 01.9.15l4.596 7.362a.53.53 0 01-.101.696 8.694 8.694 0 01-2.834 1.538c-1.549.484-2.432.463-3.316.442-.884-.021-1.768-.042-3.318.442-1.314.41-1.868.948-2.07 1.204-.048.061-.07.21-.032.279.4.724 2.295 4.06 3.794 6.693a.998.998 0 00.695.49l3.07.541a5.19 5.19 0 006.012-4.21l2.54-14.408a5.19 5.19 0 00-4.21-6.012z"
              />
              <path
                d="M16.646 13.074c-1.303.34-2.093.316-2.882.29-.79-.024-1.58-.049-2.881.291-.744.216-1.178.592-1.587 1.072a.197.197 0 00-.018.23l2.482 4.069c.064.106.209.13.307.055.374-.283 1.28-.9 2.42-1.198 1.469-.383 2.348-.375 3.273-.366.717.007 1.462.014 2.53-.16a4.454 4.454 0 001.629-.635.347.347 0 00.094-.488L19.472 12.3a.421.421 0 00-.56-.128 9.57 9.57 0 01-2.265.902z"
              />
            </g>
          </svg>
        </span>
        common.purpose.cf.continuous common.plans.subscription
      </p>
      <div
        class="Layout--horizontal StyledProps--main"
      >
        <div
          class="Layout--horizontal StyledProps--main StyledProps--flex"
        >
          <div
            class="MultiStepProgressIndicator--dot MultiStepProgressIndicator--dotSuccess MultiStepProgressIndicator--spacing"
          />
        </div>
        <div
          class="Layout--horizontal StyledProps--main StyledProps--flex"
        >
          <div
            class="MultiStepProgressIndicator--statusBar"
          >
            <div
              class="MultiStepProgressIndicator--bar MultiStepProgressIndicator--fullBar"
            />
            <div
              class="MultiStepProgressIndicator--bar MultiStepProgressIndicator--barSuccess MultiStepProgressIndicator--fullBar"
            />
          </div>
          <div
            class="MultiStepProgressIndicator--dot MultiStepProgressIndicator--dotSuccess MultiStepProgressIndicator--spacing"
          />
        </div>
        <div
          class="Layout--horizontal StyledProps--main StyledProps--flex"
        >
          <div
            class="MultiStepProgressIndicator--statusBar"
          >
            <div
              class="MultiStepProgressIndicator--bar MultiStepProgressIndicator--fullBar"
            />
          </div>
          <div
            class="MultiStepProgressIndicator--dot MultiStepProgressIndicator--dotOutLine MultiStepProgressIndicator--spacing"
          />
        </div>
      </div>
      <p
        class="StyledProps--font StyledProps--main StyledProps--padding-top-large StyledProps--font-variation-h3"
      >
        authSettings.billing.step
      </p>
    </div>
    <div
      class="Layout--horizontal StyledProps--main footer"
    >
      <div
        class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
      >
        <button
          class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--withLeftIcon Button--variation Button--variation-secondary"
          type="button"
        >
          <span
            class="bp3-icon bp3-icon-chevron-left StyledProps--main"
            icon="chevron-left"
          >
            <svg
              data-icon="chevron-left"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                chevron-left
              </desc>
              <path
                d="M7.41 8l3.29-3.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71l-4 4C5.11 7.47 5 7.72 5 8c0 .28.11.53.29.71l4 4a1.003 1.003 0 001.42-1.42L7.41 8z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          <span
            class="bp3-button-text"
          >
            back
          </span>
        </button>
        <button
          class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--withRightIcon Button--variation Button--variation-primary"
          type="button"
        >
          <span
            class="bp3-button-text"
          >
            authSettings.billing.next
          </span>
          <span
            class="bp3-icon bp3-icon-chevron-right StyledProps--main"
            icon="chevron-right"
          >
            <svg
              data-icon="chevron-right"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                chevron-right
              </desc>
              <path
                d="M10.71 7.29l-4-4a1.003 1.003 0 00-1.42 1.42L8.59 8 5.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l4-4c.18-.18.29-.43.29-.71 0-.28-.11-.53-.29-.71z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        class="Layout--vertical StyledProps--main"
      >
        <p
          class="StyledProps--font StyledProps--main"
        >
          authSettings.costCalculator.payingToday common.perMonth
        </p>
        <p
          class="StyledProps--font StyledProps--main"
        >
          authSettings.plusTax
        </p>
      </div>
    </div>
  </div>
</div>
`;
